scripts/ado-terraform-nagger.py

```diff
@@ -0,0 +1,233 @@
+import os
+import re
+import sys
+import json
+import logging
+import argparse
+import subprocess
+from packaging import version
+from json.decoder import JSONDecodeError
+
+# Global variable used to exit with error at the end of all checks.
+# To be updated from default value by logging function.
+errors_detected = False
+
+parser = argparse.ArgumentParser(description="ADO Terraform version nagger")
+parser.add_argument(
+    "-d",
+    "--debug",
+    help="Show debug logs",
+    action="store_const",
+    dest="loglevel",
+    const=logging.DEBUG,
+    default=logging.INFO,
+)
+args = parser.parse_args()
+
+logging.basicConfig(
+    level=args.loglevel,
+    format="[%(asctime)s] {%(filename)s:%(lineno)d} %(levelname)s: %(message)s",
+    handlers=[
+        logging.StreamHandler(stream=sys.stdout),
+    ],
+)
+logger = logging.getLogger()
+
+semver_regex = (
+    "\\s*(?:v\\.?)?(?P<major>\\d+)\\. (?P<minor>\\d+)\\. (?P<patch>\\d+)"
+    "(?:-(?P<pre>(?:[0-9A-Za-z-]|[1-9A-Za-z-][0-9A-Za-z-]*)"
+    "(?:\\.[0-9A-Za-z-]|[1-9A-Za-z-][0-9A-Za-z-]*)*))?"
+    "(?:\\+(?P<meta>[0-9A-Za-z-]+(?:\\.[0-9A-Za-z-]+)*))?\\s*$"
+)
+
+def load_file(filename):
+    """
+    This function loads a file from the same directory as the script itself.
+    Args:
+        filename (str): The name of the file to load.
+    Returns:
+        str: The contents of the file.
+    Raises:
+        FileNotFoundError: If the specified file does not exist.
+    """
+    # Get the path of the script
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+    # Construct the path to the file
+    file_path = os.path.join(script_dir, filename)
+    # Open and read the file
+    try:
+        with open(file_path, 'r') as f:
+            contents = f.read()
+    except FileNotFoundError:
+        raise FileNotFoundError(f"The file '{filename}' does not exist.")
+    return contents
+
+def log_message(message_type, message):
+    """
+    This function logs a given message with the logging library and,
+    if the system is running in Azure DevOps
+    (as determined by the SYSTEM_PIPELINESTARTTIME environment variable),
+    it also logs a warning issue with Azure DevOps. If message_type is set to warning
```

Comment:
How will people get notified about warnings?
From what I see at the moment it shows in the checks view:
![image](https://user-images.githubusercontent.com/21194782/222409063-07fd657a-5208-41f5-ba84-3a72db920fb3.png)

and in the summary view in ADO:
![image](https://user-images.githubusercontent.com/21194782/222409133-376ebb22-398e-4669-82f5-38ac63529b78.png)

but if the build is passing or they go straight to the logs they would never see this from what I can tell unless I'm missing something?

Reply:
So the warnings are displayed both in the output of the job run as well as the UI (with errors displayed directly on PR azure pipelines output).
There is no implementation that would notify certain person via external channel if that's what you're asking ?

Reply:
yes it is what I'm asking 👍

Reply:
OK, that functionality should now be added via https://github.com/hmcts/cnp-azuredevops-libraries/pull/137/commits/0721b02740e8a897292fa387e59d26b6027b3471